base/sparse/sparsematrix.jl

```diff
@@ -1564,10 +1602,249 @@ function _map_notzeropres!{Tf}(f::Tf, fillvalue, C::SparseMatrixCSC, A::SparseMa
     end
     return C
 end
+# _broadcast_zeropres!/_broadcast_notzeropres! specialized for a pair of (input) sparse matrices
+function _broadcast_zeropres!{Tf}(f::Tf, C::SparseMatrixCSC, A::SparseMatrixCSC, B::SparseMatrixCSC)
+    isempty(C) && (fill!(C.colptr, 1); return C)
+    spaceC = min(length(C.rowval), length(C.nzval))
+    rowsentinelA = convert(eltype(A.rowval), C.m + 1)
```

Comment:
probably just `eltype(A.rowval)(C.m + 1)`, since that does a type assertion to ensure the compiler knows the type of the result, unlike `convert`.   Though I wonder if it is better to just use `typemax(Int)` as the sentinal in all cases?